src/screens/PostScreen.tsx

```diff
@@ -88,31 +88,39 @@ const styles = StyleSheet.create({
 export default function PostScreen() {
 	const [selectedImage, setSelectedImage] = useState('');
 	const [selectedImageId, setSelectedImageId] = useState('');
-	const [loading, setLoading] = useState(false);
 	const [postDescription, setPostDescription] = useState('');
-	const [pointsData, setPointsData] = useState<PointToClaim[]>([]);
+	const { isLoading, data: pointsData } = usePointToClaimQuery();
 	const [modalVisible, setModalVisible] = useState(false);
-	const [selectedPoint, setSelectedPoint] = useState(null);
+	const [selectedPoint, setSelectedPoint] = useState<number | null>(null);
 	const selectedPointData = pointsData?.find((item) => item.id === selectedPoint);
 	const { session } = useAppContext();
+	const createPostMutation = useCreateUserPostsMutation();
+	const createPointMutation = useCreateUserPointsMutation();
 
 	const createPost = async () => {
 		const imageUrl = await getImageUrl(selectedImageId);
 
 		// Create post
-		const { error } = await supabase.from('post').insert({
-			id: 55, // @TODO fix supabase not generating serial ids or use uuid
-			title: 'Post test',
-			description: postDescription,
-			user_id: session?.user.id,
-			image_url: imageUrl,
+		const post = await createPostMutation.mutateAsync({
+			id: session?.user.id as string,
```

Comment:
This is automatically generated in Supabase and is an integer, so can be removed

Reply:
This is the user_id, which is a GUID (string):
<img width="651" alt="image" src="https://user-images.githubusercontent.com/55079408/209725298-4d5cd336-5540-4ca7-83d4-f42aba1c30e6.png">